### DISCUSSION TITLE
Port number arrangement

### DISCUSSION THREAD

**Discussion by seokho-son** [m-cmp/m-cmp#12] [General]
## 각 서브시스템별 활용 서버의 포트를 재조정하기 위한 논의

- 각 서브시스템별 현황 확인 요청 (활용하는 외부 서버 포함)
- 정책 수립
  - 주요 오픈소스 활용 포트 확인
  - 서브시스템 구성 쉽게 인지할 수 있는 방안 제시 등

**Comment by seokho-son** [comment]
현재 각 서브시스템에서 활용하고 있는 포트 조사. (+선호하는 포트가 있는 경우 공유)

**Comment by seokho-son** [comment]
@MZC-CSC  @raccoon-mh @SungWoongz @hippo (cc: @innodreamer  )
 https://github.com/m-cmp/mc-iam-manager :

- 5000 (mc-iam-manager)
- 5432 (db(postgres))
- 443 (keycloak-포트변경불가)

 https://github.com/m-cmp/mc-web-console :

- 3000 (mc-web-console API)
- 3001 (mc-web-console front)

 https://github.com/m-cmp/mc-admin-cli :

- 대상없음

---

@pbcccbeatboard-strato (cc: @sykim-etri )
 https://github.com/m-cmp/mc-workflow-manager :

- 18083 (mc-workflow-manager)
- 80, 443 (webUI)
- 8080 (jenkins)

 https://github.com/m-cmp/mc-application-manager :

- 18084 (mc-application-manager)
- 80, 443 (webUI)
- 8081 (nexus)

---

@heedaeshin (cc: @yunkon-kim )
 https://github.com/m-cmp/mc-data-manager :
- 3300 (mc-data-manager)

---

@hyeon-inno @yunha-inno 
 https://github.com/m-cmp/mc-observability :

- 18080 (Observability-Manager)
- 18081 (Observability-Agent(-Manager))
- 8086 (InfluxDB)
- 3306 (MariaDB)
- 8888 (Chronograf)

---

@BSjaeyoung-kim 
 https://github.com/m-cmp/mc-cost-optimizer :
- 8080 (portal Frontend)
- 9090 (portal Backend)
- 8081~8083 (cost collect Batch)
- 9000 (Alarm/slack service)
- 3306 (Portal DB)

---

@seokho-son
 https://github.com/m-cmp/mc-infra-manager (submodule: CB-Tumblebug) :
- 1323 (CB-TB)
- 1324 (CB-MapUI)
- 2379, 2380 (ETCD)

---

@powerkimhub    
 https://github.com/m-cmp/mc-infra-connector (submodule: CB-Spider) : 
- 1024 (Rest Server)
- 2048 (gRPC for CLI, deprecated soon)

**Reply by BSjaeyoung-kim** [reply]
https://github.com/m-cmp/mc-cost-optimizer :

portal Frontend : 8080
portal Backend : 9090
cost collect Batch : 8081~8083
Alarm/slack service : 9000
Poral DB : 3306

**Reply by heedaeshin** [reply]
https://github.com/m-cmp/mc-data-manager :

data manager : 3300

**Reply by seokho-son** [reply]
@heedaeshin 안녕하세요. 답변 감사합니다. 
mc-data-manager 에 기본적인 API 서버, 웹 제공을 위한 서버 등이 있을 것으로 예상되는데, (https://github.com/m-cmp/mc-data-manager/blob/main/cmd/server.go#L29)
현재 활용하는 port가 전혀 없는 것이 맞나요? 곧 제공할 예정인 경우라도 대략적으로 공유해주시면 좋을 것 같습니다.

**Reply by heedaeshin** [reply]
@seokho-son 
선호하는 포트가 없으므로 None으로 작성하였습니다.
지정포트가 필요하시면 3300 번으로 기록하겠습니다.

**Reply by hyeon-inno** [reply]
https://github.com/m-cmp/mc-observability :

Observability-Manager : 18080
Observability-Agent(-Manager) : 18081
InfluxDB : 8086
MariaDB : 3306

** Chronograf : 8888

**Reply by raccoon-mh** [reply]
https://github.com/m-cmp/mc-iam-manager :
- mciammanager = :5000
  - keycloak = :443 (IDP 등록용도로 CSP에서 요구하는 https 포트)
  - db(postgres) = :5432

https://github.com/m-cmp/mc-web-console :
- webconsole API = :3000
- webconsole front = :3001

**Reply by seokho-son** [reply]
@raccoon-mh `keycloak = :443 (IDP 등록용도로 CSP에서 요구하는 https 포트)` 혹시 컨테이너로 구동하는 경우에도 호스트의 포트가 443 이어야 하나요? -p 443:443 (?)

**Reply by raccoon-mh** [reply]
@seokho-son 
맞습니다. 
CSP 에서 IDP 등록하는 절차에 AWS 및 Alibaba의(현재 PoC 가 완료된 CSP) 경우
 https;// 로 시작하고 포트 주소가 없는 well-known/openid-configuration 주소를 요구하고 있습니다.
 
 AWS 자격증명 공급자 구성 검증 관련 Docs 링크 참조 : https://docs.aws.amazon.com/ko_kr/IAM/latest/UserGuide/id_roles_providers_create_oidc.html#manage-oidc-provider-prerequisites

**Reply by raccoon-mh** [reply]
추가로 외부 포트만 443 으로 노출하면되고, 내부에서는 설정을 통해 포트 변경은 가능합니다.

**Comment by seokho-son** [comment]
### FYI 포트를 중심으로 정렬한 유명한 오픈소스 소프트웨어 목록 (정확한 내용은 아님 참고)

| 포트 번호 | 소프트웨어                                             |
|-----------|-------------------------------------------------------|
| 22        | GitLab                                                |
| 80        | Apache, Nginx, GitLab, Joomla, Drupal, Contour        |
| 389       | OpenLDAP                                              |
| 443       | Apache, Nginx, GitLab, Nextcloud, Træfik, Joomla, Drupal, Contour |
| 636       | OpenLDAP                                              |
| 2181      | Zookeeper                                             |
| 2375      | Docker                                                |
| 2376      | Docker                                                |
| 2379      | Etcd                                                  |
| 2380      | Etcd                                                  |
| 3000      | Grafana, Metabase, Rocket.Chat, Jupyter, Chronograf, Kibana, Webmin, Gogs |
| 3306      | MySQL                                                 |
| 33060     | MySQL X Protocol                                      |
| 4194      | Kubernetes cAdvisor                                   |
| 5000      | OpenStack Keystone                                    |
| 5001      | OpenStack Swift                                       |
| 5432      | PostgreSQL                                            |
| 5601      | Kibana                                                |
| 5672      | RabbitMQ                                              |
| 5984      | CouchDB                                               |
| 6080      | OpenStack Horizon                                     |
| 6379      | Redis                                                 |
| 6443      | Kubernetes API Server                                 |
| 7000      | OpenStack Cinder, Cassandra                           |
| 7001      | OpenStack Cinder, Cassandra                           |
| 7077      | Apache Spark                                          |
| 7474      | Neo4j                                                 |
| 7687      | Neo4j Bolt                                            |
| 8000      | ZooNavigator, Kong, Django, Varnish, Kubernetes Dashboard |
| 8001      | Kong                                                  |
| 8080      | Tomcat, Keycloak, Jenkins, Presto, Apache NiFi, Airflow, Tyk, SonarQube, Nexus, OpenShift, Spark, Spring Boot, WildFly, JBoss EAP, Ghost, Odoo, Pi-hole, Rundeck, Apache Traffic Server, Caddy, Kubernetes Dashboard, Linkerd |
| 8081      | Nexus, Flink, Artifactory                             |
| 8082      | Druid                                                 |
| 8086      | InfluxDB                                              |
| 8088      | Apache Superset                                       |
| 8091      | Couchbase Web Administration                          |
| 8123      | ClickHouse                                            |
| 8140      | Puppet                                                |
| 8161      | ActiveMQ                                              |
| 8200      | Vault                                                 |
| 8332      | Bitcoin Core RPC                                      |
| 8333      | Bitcoin Core P2P                                      |
| 8443      | OpenShift, Spring Boot, WildFly, JBoss EAP, Linkerd   |
| 8500      | Consul, Kubernetes Node Port Range                    |
| 8600      | Consul DNS                                            |
| 8888      | Jupyter Notebook, MoinMoin                            |
| 9000      | SonarQube, Graylog, MinIO, Spinnaker, Spark, Play Framework, Sentry, Zabbix, Caddy            |
| 9042      | Cassandra                                             |
| 9090      | Prometheus, HBase REST                                |
| 9091      | HBase Thrift                                          |
| 9092      | Kafka                                                 |
| 9100      | Prometheus Node Exporter                              |
| 9200      | Elasticsearch                                         |
| 9300      | Elasticsearch                                         |
| 9418      | Git                                                   |
| 9990      | WildFly Management                                    |
| 10250     | Kubernetes Kubelet API                                |
| 10255     | Kubernetes Kubelet Read-Only API                      |
| 15000     | Istio Proxy Admin                                     |
| 15021     | Istio Pilot HTTP                                      |
| 15090     | Istio Proxy Telemetry                                 |
| 15672     | RabbitMQ Management                                   |
| 27017     | MongoDB                                               |
| 27018     | MongoDB (Secondary)                                   |
| 27019     | MongoDB (Arbiter)                                     |
| 28017     | MongoDB Web Status                                    |
| 30000     | MongoDB (Sharded Cluster)                             |
| 40000     | MongoDB (Sharded Cluster)                             |

**Comment by seokho-son** [comment]
admin-cli 관점에서도 혹시 내부 컴포넌트가 활용하는 포트 번호 (재)조정에 의견이 있으시면 공유 부탁드립니다.
@MZC-CSC @dev4unet

**Comment by pbcccbeatboard-strato** [comment]
Application-manager 18084
Workflow-manager 18083
webUI 80, 443
jenkins 8080
nexus 8081

사용하고 있습니다.

현재 사용하지 않으려 하나 혹시 모르는(?) oss인 argoCD 80, 443(타 구성요소 포트는 생략)입니다.

선호하는 포트는 없습니다.
